msgraph/synchronization.go

```diff
@@ -0,0 +1,318 @@
+			HasTenantId: true,
+		},
+	})
+	if err != nil {
+		return nil, status, fmt.Errorf("SynchronizationJobClient.BaseClient.Get(): %v", err)
+	}
+
+	defer resp.Body.Close()
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, status, fmt.Errorf("io.ReadAll(): %v", err)
+	}
+
+	var synchronizationJob SynchronizationJob
+	if err := json.Unmarshal(respBody, &synchronizationJob); err != nil {
+		return nil, status, fmt.Errorf("json.Unmarshal(): %v", err)
+	}
+
+	return &synchronizationJob, status, nil
+}
+
+// GetSecrets retrieves a SynchronizationSecret
+func (c *SynchronizationJobClient) GetSecrets(ctx context.Context, servicePrincipalId string) (*SynchronizationSecret, int, error) {
+	resp, status, _, err := c.BaseClient.Get(ctx, GetHttpRequestInput{
+		ValidStatusCodes: []int{http.StatusOK},
+		Uri: Uri{
+			Entity:      fmt.Sprintf("/servicePrincipals/%s/synchronization/secrets/", servicePrincipalId),
+			HasTenantId: true,
+		},
+	})
+	if err != nil {
+		return nil, status, fmt.Errorf("SynchronizationJobClient.BaseClient.Get(): %v", err)
+	}
+
+	defer resp.Body.Close()
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, status, fmt.Errorf("io.ReadAll(): %v", err)
+	}
+
+	var synchronizationSecret SynchronizationSecret
+	if err := json.Unmarshal(respBody, &synchronizationSecret); err != nil {
+		return nil, status, fmt.Errorf("json.Unmarshal(): %v", err)
+	}
+
+	return &synchronizationSecret, status, nil
+}
+
+// Adds a SynchronizationSecrets.
+func (c *SynchronizationJobClient) SetSecrets(ctx context.Context, synchronizationSecret SynchronizationSecret, servicePrincipalId string) (int, error) {
+	var status int
+
+	body, err := json.Marshal(synchronizationSecret)
+	if err != nil {
+		return status, fmt.Errorf("json.Marshal(): %v", err)
+	}
+	resp, status, _, err := c.BaseClient.Put(ctx, PutHttpRequestInput{
+		Body:             body,
+		ValidStatusCodes: []int{http.StatusNoContent},
+		Uri: Uri{
+			Entity:      fmt.Sprintf("/servicePrincipals/%s/synchronization/secrets", servicePrincipalId),
+			HasTenantId: true,
+		},
+	})
+	if err != nil {
+		return status, fmt.Errorf("SynchronizationJobClient.BaseClient.Put(): %v", err)
+	}
+
+	defer resp.Body.Close()
+	_, err = io.ReadAll(resp.Body)
+	if err != nil {
+		return status, fmt.Errorf("io.ReadAll(): %v", err)
+	}
+
+	return status, nil
+}
+
+// Creates a SynchronizationJob.
+func (c *SynchronizationJobClient) Create(ctx context.Context, synchronizationJob SynchronizationJob, servicePrincipalId string) (*SynchronizationJob, int, error) {
+	var status int
+
+	body, err := json.Marshal(synchronizationJob)
+	if err != nil {
+		return nil, status, fmt.Errorf("json.Marshal(): %v", err)
+	}
+	resp, status, _, err := c.BaseClient.Post(ctx, PostHttpRequestInput{
+		Body:             body,
+		ValidStatusCodes: []int{http.StatusCreated},
+		Uri: Uri{
+			Entity:      fmt.Sprintf("/servicePrincipals/%s/synchronization/jobs/", servicePrincipalId),
```

Comment:
```suggestion
			Entity:      fmt.Sprintf("/servicePrincipals/%s/synchronization/jobs", servicePrincipalId),
```

Reply:
Fixed in b574351989a455ba92195ee57c8fc6a5008ecbba